namerd/storage/zk/src/main/scala/com/twitter/finagle/serverset2/buoyant/ZkDtabStore.scala

```diff
@@ -0,0 +1,127 @@
+package com.twitter.finagle.serverset2.buoyant
+
+import com.twitter.finagle.Dtab
+import com.twitter.finagle.serverset2.client.KeeperException.{BadVersion, NoNode}
+import com.twitter.finagle.serverset2.client._
+import com.twitter.finagle.serverset2.{RetryStream, Zk2Resolver, ZkSession => FZkSession}
+import com.twitter.finagle.stats.DefaultStatsReceiver
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.io.Buf
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.namerd.DtabStore.{DtabNamespaceAlreadyExistsException, DtabNamespaceDoesNotExist, DtabVersionMismatchException}
+import io.buoyant.namerd.{DtabStore, VersionedDtab}
+import java.nio.ByteBuffer
+
+/**
+ * A DtabStore which stores dtabs in ZooKeeper.  Dtabs are stored, Utf8
+ * encoded, in the ZooKeeper node /dtabs/<ns>.
+ */
+class ZkDtabStore(
+  hosts: String,
+  zkPrefix: String,
+  sessionTimeout: Option[Duration]
+) extends DtabStore {
+
+  private[this] val log = Logger()
+  private[this] val retryStream = RetryStream()
```

Comment:
what does this mean, practically? Should this be configurable?  Should this be an exponentialJitter backoff?